### PR TITLE
Add test for FVFluxBCs possible issues

### LINKED ISSUE #16882: Error messages when FVFluxBCs are not well placed
## Reason
If you place FVFluxBCs in odd locations, like an internal sideset or a sideset that does not have the corresponding finite volume variable, you will be greated with "should not get here" ^^. 

## Design
Have distinct message for the two possibilities, and give enough information for the user to figure it out. 

## Impact
Catch some recent block restriction issues @smharper ran in

### PR DESCRIPTION
**Pull request by GiudGiud**
Closes #16882

Cant figure out the exact format for the error tests yet. I will tomorrow

### CODE REVIEW

**Comment by GiudGiud** [comment]
If it doesnt error out, this should be ready for review @lindsayad

File: test/tests/fvbcs/fv_neumannbc/tests [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    exodiff = 'fv_neumannbc_out.e'
+    requirement = 'The system shall run a simple 1D diffusion problem with a Neumann BC value.'
+  []
+  [fvbcs_internal]
+    design = 'FVNeumannBC.md'
+    issues = '#16882'
+    type = RunException
+    input = 'fv_neumannbc.i'
+    cli_args = "FVBCs/inactive=''"
+    expect_err = 'A FVFluxBC is being triggered on an internal face with centroid'
```
Comment (lindsayad) [inline test/tests/fvbcs/fv_neumannbc/tests:16]:
```suggestion
    expect_err = 'A FVFluxBC is being triggered on an internal face'
```

Comment (lindsayad) [reply test/tests/fvbcs/fv_neumannbc/tests:16]: I don't know if I love the current message with the centroid reporting, but I don't have anything better right now. However, in the future I could see us modifying the error a bit and maybe not printing the centroid (e.g. we might just print the adjoining subdomain ids or something)

File: test/tests/fvbcs/fv_neumannbc/tests [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    requirement = 'The system shall error out if FVFluxBC, in this case a FVNeumannBC, are used internally.'
```
Comment (lindsayad) [inline test/tests/fvbcs/fv_neumannbc/tests:17]: We try to keep implementation details like object names out of requirements. So maybe `FVFluxBC` -> `finite volume flux boundary condition` and `FVNeumannBC` -> `Neumann boundary condition`.

File: test/tests/fvbcs/fv_neumannbc/tests [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  []
+  [fvbcs_disconnected_from_variable]
+    design = 'FVNeumannBC.md'
+    issues = '#16882'
+    type = RunException
+    input = 'fv_neumannbc.i'
+    expect_err = 'A FVFluxBC is being triggered on a face which does not connect to a block with the relevant finite volume variable.'
+    cli_args = "Mesh/mesh/subdomain_id='1 2'"
+    requirement = 'The system shall error out if FVFluxBC are used on a mesh face that is not connected to an element with the corresponding finite volume variable.'
```
Comment (lindsayad) [inline test/tests/fvbcs/fv_neumannbc/tests:26]: As above

**Review by lindsayad — CHANGES_REQUESTED** [review]
small suggestions

File: test/tests/fvbcs/fv_neumannbc/tests (hunk 1 above)
Comment (GiudGiud) [reply test/tests/fvbcs/fv_neumannbc/tests]: I would be fine with adjoining subdomain id, I just would hate to have the surface id, which is harder to track down.

**Review by lindsayad — APPROVED** [review]

**Pull request merged**